### PR TITLE
Reverse a Linked List in golang

### PR DESCRIPTION
**Pull request by nick-leslie**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

yes

### Description

A program to reverse a singly linked list in golang.

### Checklist

- [yes] I've read the contribution guidelines.
- [yes] I've checked the issue list before deciding what to submit.
- [yes] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

Issue#1620

### CODE REVIEW

File: Go/ds/singly_linked_list_reversed.go [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+
+func main() {
+	list := makeLinkedList()
+	userInput := 0
+	shouldRun := true
+	//I used the format from the other linked list in go to keep consistency between the two programs
+	for shouldRun == true {
+		fmt.Printf("1 -> insert element at front\n2 -> insert dummy data\n" +
+			"3 -> print linked list\n4 -> reverse linked list\n" +
+			"5 -> quit\n")
```
Comment (HarshCasper) [inline Go/ds/singly_linked_list_reversed.go]: Can you use better Console Statements?

File: Go/ds/singly_linked_list_reversed.go [hunk 2]
```diff
@@ -0,0 +1,104 @@
+package main
```
Comment (HarshCasper) [inline Go/ds/singly_linked_list_reversed.go:4]: Add a beginner pitch about the Code/Algorithm/Data Structures

File: Go/ds/singly_linked_list_reversed.go [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+		currentNode.next = prev
+		prev = currentNode
+		currentNode = next
+	}
+	list.tail = list.head
+	list.head = prev
+	fmt.Println("successfully reversed the linked list")
+}
+
+//fillListWithDummyData: add values 1 - 9 to the linked list for testing reasons
```
Comment (HarshCasper) [inline Go/ds/singly_linked_list_reversed.go]: Please use better comments to exactly show what is being done, rather than vague idea about the same

**Review by HarshCasper — CHANGES_REQUESTED** [review]

File: Go/ds/singly_linked_list_reversed.go [hunk 4]
```diff
@@ -0,0 +1,120 @@
+/*
+ reverce linked list is an implimention of a singaly linked list with an algorithm to reverce the order of the list
```
Comment (HarshCasper) [inline Go/ds/singly_linked_list_reversed.go]: Typo. It should be `reverse`

**Review by HarshCasper — CHANGES_REQUESTED** [review]
Tiny change required. Please squash your Commits as well. I will merge it soon after

**Comment by nick-leslie** [comment]
could you check if I squashed my commits correctly? it's my first time doing so and I don't know if I did it correctly .

**Comment by atarax665** [comment]
> could you check if I squashed my commits correctly? it's my first time doing so and I don't know if I did it correctly.

Your commits have not been squashed yet.

**Comment by nick-leslie** [comment]
My commits should be squashed.

**Review by HarshCasper — APPROVED** [review]

**Pull request merged**